pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -354,9 +354,28 @@ func ReadLogs(path, containerID string, opts *LogOptions, runtimeService interna
 	}
 }
 
+func isPodRunning(id string, r internalapi.RuntimeService) (bool, error) {
+	s, err := r.ContainerStatus(id)
+	if err != nil {
+		return false, err
+	}
+	// Only keep following container log when it is running.
+	if s.State != runtimeapi.ContainerState_CONTAINER_RUNNING {
```

Comment:
The container may in starting state, use `s.State == runtimeapi.ContainerState_CONTAINER_EXITED` instead?

Reply:
I was just reusing the logic that already existed there as to not complicate things.